### ISSUE TITLE
Support providing authentication details when using ntfy:// links

### ISSUE THREAD

**Issue by laundmo** [binwiederhier/ntfy#597]
We'd like to use ntfy to notify end users, but getting them to configure the app is a lot of work to explain and get users to do.

Therefore, it would be really convenient to be able to provide users with ntfy:// links, which include the authentication details needed to actually subscribe to the topic.

Currently, if we use a ntfy:// link to a protected topic, it will register fine and will be stuck on `reconnecting...`constantly.

Proposed solutions:

- UI popup asking for username/password when a ntfy:// link to a protected topic is used
- ability to provide username/password inside a ntfy:// link (preferred)

Thank you for this project.

**Comment by piwi82** [comment]
Therefore, users could scan a QR code to subscribe 📱
Making **ntfy** more accessible to end users will drive adoption 😎

**Comment by dgtlmoon** [comment]
theres a PR for android over at https://github.com/binwiederhier/ntfy-android/pull/67

**Comment by vovayartsev** [comment]
A very wanted feature indeed!
💯 this great app could be used as a general reminder not only by technical users - if only the subscription setup was as easy as "scan the QR code".

Adding [Deferred Deep Linking](https://www.branch.io/glossary/deferred-deep-linking/) could automate the entire setup process to just 1 step:

![example](https://github.com/binwiederhier/ntfy/assets/204231/1c4afc91-ce9d-47d1-b3a8-4492adeca608)

Once this is in place, the officials can make their users' life so much easier in lots of annoying "in-real-world" scenarios:
* tracking a mail delivery    
* tracking visa application process
* queueing in a hospital/bank/etc  
* waiting for the order in McDonalds   (QR code can be on a receipt)

**Comment by FAUSheppy** [comment]
This would be a great feature!

**Comment by binwiederhier** [comment]
This is very easy to add, but do people really want the ability to add username and password in a QR code? That seems like a recipe for disaster.

```ntfy://ntfy.sh/mytopic?username=phil&password=ThisSeemsSafe```

<img width="510" height="513" alt="Image" src="https://github.com/user-attachments/assets/cf498070-5391-4388-b032-0751291e5af5" />

**Comment by laundmo** [comment]
Yes(ish)

Its a better solution than having to teach non-technical users how to use ntfy, but the best solution that isn't unsafe would probably be single-use links. But that requires the app communicate with the server, and the server to expose a new unauthenticated endpoint.

Really, some way of making it easier for non-technical users to set up their phone to recieve private ntfy notifications would be great.

**Comment by binwiederhier** [comment]
Never in a million years would I have thought that non-technical users would want to use ntfy when I started making it. ntfy is about as nerdy as it gets, haha. 

I get it though. It's not the first time I've heard it. 

The ntfy:// links are pretty terrible to be honest, because if you don't have the app installed and you scan a QR code, it'll not work at all. I am toying with the idea of making a page "https://link.ntfy.sh/#subscribe=..." that will display steps of what to do, so that you can scan it with your normal camera and if you have the app, ntfy will open and subscribe, but if you don't it'll show instructions.

**Comment by laundmo** [comment]
> Never in a million years would I have thought that non-technical users would want to use ntfy when I started making it. 

Well, want is relative. We want to direct our users away from using messangers as push notification platforms, and ntfy is a great alternative.

> I am toying with the idea of making a page "https://link.ntfy.sh/#subscribe=..." that will display steps of what to do, so that you can scan it with your normal camera and if you have the app, ntfy will open and subscribe, but if you don't it'll show instructions.

That would be amazing!
